### ISSUE TITLE
TargetName must match RRTYPE

### ISSUE THREAD

**Issue by squarooticus** [MikeBishop/dns-alt-svc#251]
§ 2.4.1 says:

```
The primary purpose of AliasMode is to allow aliasing at the zone apex, where
CNAME is not allowed. In AliasMode, TargetName MUST be the name of
a domain that has SVCB, AAAA, or A records. It MUST NOT be equal to
the owner name, as this would cause a loop.
```

From my reading, this should say that TargetName MUST be the name of a domain that has AAAA or A records, or records of the same RRTYPE as that of the source record. The intro says that

```
All behaviors described as applying to the SVCB RR also apply to the
HTTPS RR unless explicitly stated otherwise.
```

but that does not (and should not) explicitly encourage readers to alpha-substitute HTTPS for SVCB in every normative statement.

**Comment by bemasc** [comment]
Brian Dickson also found that line confusing, for a different reason (bottom of [this email](https://mailarchive.ietf.org/arch/msg/dnsop/_Qb7Hsn1NjTJx2UTkdS-KeYIpF0/)).  We should rephrase.

**Comment by bemasc** [comment]
This was fixed in #224
